toolkit/scripts/containerized-build/create_container_build.sh

```diff
@@ -0,0 +1,197 @@
+    if [ -d "${repo_path}/build/container-buildroot" ]; then rm -Rf ${repo_path}/build/container-buildroot; fi
+    mkdir ${repo_path}/build/container-build
+    mkdir ${repo_path}/build/container-buildroot
+    mounts="${mounts} ${repo_path}/build/container-build:${topdir}/BUILD ${repo_path}/build/container-buildroot:${topdir}/BUILDROOT"
+fi
+
+# ============ Setup tools ============
+# Copy relavant build tool executables from ${repo_path}/tools/out
+echo "Setting up tools..."
+if [[ ( ! -f "${repo_path}/toolkit/out/tools/depsearch" ) || ( ! -f "${repo_path}/toolkit/out/tools/grapher" ) || ( ! -f "${repo_path}/toolkit/out/tools/specreader" ) ]]; then build_tools; fi
+if [[ ! -f "${repo_path}/build/pkg_artifacts/graph.dot" ]]; then build_graph; fi
+cp ${repo_path}/toolkit/out/tools/depsearch ${build_dir}/
+cp ${repo_path}/toolkit/out/tools/grapher ${build_dir}/
+cp ${repo_path}/toolkit/out/tools/specreader ${build_dir}/
+cp ${repo_path}/build/pkg_artifacts/graph.dot ${build_dir}/
+
+# ========= Setup mounts =========
+echo "Setting up mounts..."
+
+mounts="${mounts} ${repo_path}/out/RPMS:/mnt/RPMS"
+if [[ "${mode}" == "build" ]]; then
+    # Add extra build mounts
+    mounts="${mounts} ${repo_path}/build/INTERMEDIATE_SRPMS:/mnt/INTERMEDIATE_SRPMS ${repo_path}/SPECS:${topdir}/SPECS"
+fi
+
+rm -f ${build_dir}/mounts.txt
+for mount in $mounts $extra_mounts; do
+    if [[ -d "${mount%%:*}" ]]; then
+        echo "${mount%%:*}' -> '${mount##*:}"  >> "${build_dir}/mounts.txt"
+    else
+        echo "WARNING: '${mount%%:*}' does not exist. Skipping mount."  >> "${build_dir}/mounts.txt"
+        continue
+    fi
+    mount_arg=" $mount_arg -v '$mount' "
+done
+
+# ============ Build the dockerfile ============
+dockerfile="${script_dir}/resources/mariner.Dockerfile"
+
+if [[ "${mode}" == "build" ]]; then # Configure base image
+    echo "Importing chroot into docker..."
+    chroot_file="${repo_path}/build/worker/worker_chroot.tar.gz"
+    if [[ ! -f "${chroot_file}" ]]; then build_chroot; fi
+    chroot_hash=$(sha256sum "${chroot_file}" | cut -d' ' -f1)
+    # Check if the chroot file's hash has changed since the last build
+    if [[ ! -f "${script_dir}/build_container/hash" ]] || [[ "$(cat "${script_dir}/build_container/hash")" != "${chroot_hash}" ]]; then
+        echo "Chroot file has changed, updating..."
+        echo "${chroot_hash}" > "${script_dir}/build_container/hash"
+        sudo docker import "${chroot_file}" "mcr.microsoft.com/cbl-mariner/containerized-rpmbuild:${version}"
+    else
+        echo "Chroot is up-to-date"
+    fi
+    container_img="mcr.microsoft.com/cbl-mariner/containerized-rpmbuild:${version}"
+else
+    container_img="mcr.microsoft.com/cbl-mariner/base/core:${version}"
+fi
+
+# ================== Launch Container ==================
+echo "Checking if build env is up-to-date..."
+docker_image_tag="mcr.microsoft.com/cbl-mariner/${USER}-containerized-rpmbuild:${version}"
```

Comment:
You don't need to specify a full URL if you are just using the container locally.

So, this can be just: `cbl-mariner/${USER}-containerized-rpmbuild:${version}` or whatever.

Reply:
Fair.